frontend/src/router/pages/PreviewChanges.jsx

```diff
@@ -0,0 +1,344 @@
+import PropTypes from "prop-types";
+import { useNavigate, useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { formatTimestamp } from "@/lib/utils";
+import { useApiGet, useApiPost } from "@/hooks/useApi";
+
+import { faPen } from "@fa-kit/icons/classic/solid";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import NavBack from "@/components/NavBack";
+import walkInCamping from "@/assets/icons/walk-in-camping.svg";
+import LoadingBar from "@/components/LoadingBar";
+import ContactBox from "@/components/ContactBox";
+import ReadyToPublishBox from "@/components/ReadyToPublishBox";
+import FlashMessage from "@/components/FlashMessage";
+import DateRange from "@/components/DateRange";
+
+import { Link } from "react-router-dom";
+
+import "./PreviewChanges.scss";
+
+function PreviewChanges() {
+  const { parkId, seasonId } = useParams();
+  const navigate = useNavigate();
+
+  const [notes, setNotes] = useState("");
+  const [readyToPublish, setReadyToPublish] = useState(false);
+  const [showFlash, setShowFlash] = useState(false);
+
+  const { data, loading, error, fetchData } = useApiGet(`/seasons/${seasonId}`);
+
+  const {
+    sendData: saveData,
+    // error: saveError, // @TODO: handle save errors
+    loading: saving,
+  } = useApiPost(`/seasons/${seasonId}/save/`);
+
+  const { sendData: approveData, loading: savingApproval } = useApiPost(
+    `/seasons/${seasonId}/approve/`,
+  );
+
+  function navigateToEdit() {
+    navigate(`/park/${parkId}/edit/${seasonId}`);
+  }
+
+  function getPrevSeasonOperatingDates(feature) {
+    const dates = feature.dateable.previousSeasonDates.filter(
+      (dateRange) => dateRange.dateType.name === "Operation",
+    );
+
+    if (dates.length === 0) {
+      return "Not available";
+    }
+    return dates.map((date) => (
+      <DateRange key={date.id} start={date.startDate} end={date.endDate} />
+    ));
+  }
+
+  function getPrevSeasonReservationDates(feature) {
+    const dates = feature.dateable.previousSeasonDates.filter(
+      (dateRange) => dateRange.dateType.name === "Reservation",
+    );
+
+    if (dates.length === 0) {
+      return "Not available";
+    }
+    return dates.map((date) => (
+      <DateRange key={date.id} start={date.startDate} end={date.endDate} />
+    ));
+  }
+
+  function getCurrentSeasoOperationDates(feature) {
```

Comment:
Typo: `Seaso`

Reply:
We could also probably make the dateType name a parameter and reuse a single function for the Operation and Reservation dates, but as long as we fix the typo it's good for now 👍